lib/AST/ASTWalker.cpp

```diff
@@ -1571,8 +1571,9 @@ Pattern *Traversal::visitTypedPattern(TypedPattern *P) {
   else
     return nullptr;
   if (!P->isImplicit())
-    if (doIt(P->getTypeLoc()))
-      return nullptr;
+    if (auto *TR = P->getTypeRepr())
+      if (doIt(TR))
+        return nullptr;
```

Comment:
I guess this is going to be the big difference: we won't walk to the TypeLoc here. Hopefully nothing is relying on that, especially as we move more in this direction.

Reply:
I can always tell it to walk into the typeloc that's created in getTypeLoc -- should it?

Reply:
I don't see a single walkToTypeLoc(Pre|Post) in our repo that would care, so I think you should just go ahead with this.